pkg/arvo/sys/hoon.hoon

```diff
@@ -1540,6 +1540,34 @@
     ::
     a(r $(a r.a))
   ::
+  ++  jub                                               ::  add or modify
+    ~/  %jub
+    |*  [key=_?>(?=(^ a) p.n.a) fun=$-((unit _?>(?=(^ a) q.n.a)) _?>(?=(^ a) q.n.a))]
+    ^+  a
+    ::
+    ?~  a  [key (fun ~) ~ ~]
```

Comment:
Both +jub and +jib worked for me if I changed `[key (fun ~) ~ ~]` to `[key^(fun ~) ~ ~]`

Reply:
Yes, the `node` type in `tree` for `map` must be a `pair`.

PS - please make it `(pair key (fun ~))` or `[key (fun ~)]`; I hate the `+` `/` and `^` sugars so much...